Preselect all capture agent inputs when scheduling new event

Arnei opened the pull request:
Aims to fix #941.

When scheduling a new event, if the selected capture agent has any inputs, the inputs are all preselected per default now. At least one input must remain selected. Input name should be rendered nicer now.

Note that these changes are only for when scheduling a new event (except the input name rendering), because the linked issue only talked about scheduling new events. For now I would leave the discussion on how this should work for existing events to a separate issue/pr.  

For testing this, a capture agent with inputs must be configured in your Opencast. If you do not have a "real" capture agent, a dummy capture agent can still be configured. See the Opencast docs for more info: https://docs.opencast.org/develop/developer/#architecture/capture-agent/capture-agent-input/#setting-input-options

Quick showcase of the PR in action:
[Bildschirmaufzeichnung vom 2024-11-12 11-05-41.webm](https://github.com/user-attachments/assets/f49f9086-f95e-4cd0-8e40-fce43d465e2e)

snoesberger commented:
I tested this with an Extron SMP (2 inputs, channel A and B) and a pyCA (no input) and it worked as expected for both.

I only have a few minor comments:
- When no input is selected, the next button is greyed out and not clickable, but there is no indication for the user to know what is missing. Can you add some information to indicate that at least one channel needs to be selected? Perhaps a text message like the one that appears when you select an end date that is in the past?
- If the CA has inputs, the inputs field should also have a red star symbol like the other required fields.

Perhaps someone who has CAs from other vendors can have a look and see if this works well for them too?

lkiesow commented:
I'm wondering what our expectations for capture agents are for this.
Our documentation states that it's up to the agent to interpret whatever value they get.
Which seems to me like no selection would also be a valid option.

In that sense, we should at least test this with all commonly used agents before merging this:

- [ ] Extron SMP
- [ ] Adena Arec
- [ ] Epiphan Perl
- [ ] pyCA
- [ ] Galicaster *←(not sure if anyone is still using that)*

Or was this behavior already present in the old admin interface?

gregorydlogan commented:
@JamesUoM @NadiaUoM you guys still using Galicaster?

NadiaUoM commented:
Hi Greg,

Yes, we still use galicaster.

Thanks & Regards,
Nadia Malek.
From: Greg Logan ***@***.***>
Sent: 18 November 2024 23:35
To: opencast/opencast-admin-interface ***@***.***>
Cc: Nadia Malek ***@***.***>; Mention ***@***.***>
Subject: Re: [opencast/opencast-admin-interface] Preselect all capture agent inputs when scheduling new event (PR #979)


@JamesUoM [github.com]<https://urldefense.com/v3/__https:/github.com/JamesUoM__;!!PDiH4ENfjr2_Jw!AzI53ToNo0gESJRxIH9oIC99O4aSFwC0W_F-5-sJMrxvcefLcTqOI-0ksvoAeR8X1wwrZveF1spFqsmsQnUwHGuXSCBDbivd$> @NadiaUoM [github.com]<https://urldefense.com/v3/__https:/github.com/NadiaUoM__;!!PDiH4ENfjr2_Jw!AzI53ToNo0gESJRxIH9oIC99O4aSFwC0W_F-5-sJMrxvcefLcTqOI-0ksvoAeR8X1wwrZveF1spFqsmsQnUwHGuXSDswQyB0$> you guys still using Galicaster?

-
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.com/v3/__https:/github.com/opencast/opencast-admin-interface/pull/979*issuecomment-2484388788__;Iw!!PDiH4ENfjr2_Jw!AzI53ToNo0gESJRxIH9oIC99O4aSFwC0W_F-5-sJMrxvcefLcTqOI-0ksvoAeR8X1wwrZveF1spFqsmsQnUwHGuXSD4s4efZ$>, or unsubscribe [github.com]<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/BFC3B5YBSO7U6YHSBRJJBAD2BJ2TZAVCNFSM6AAAAABRTWWKAKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBUGM4DQNZYHA__;!!PDiH4ENfjr2_Jw!AzI53ToNo0gESJRxIH9oIC99O4aSFwC0W_F-5-sJMrxvcefLcTqOI-0ksvoAeR8X1wwrZveF1spFqsmsQnUwHGuXSKsgPZor$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

JamesUoM commented:
... But we are going to move to pyCA by the next academic year.

James


--
------------------------------------------------------------------------
  James S. Perrin

  Media Technologies Team
  Office One, Kilburn Building,
  The University of Manchester
  Oxford Road, Manchester, M13 9PL

  e: ***@***.***
________________________________
From: NadiaUoM ***@***.***>
Sent: 19 November 2024 10:38
To: opencast/opencast-admin-interface ***@***.***>
Cc: James Perrin ***@***.***>; Mention ***@***.***>
Subject: Re: [opencast/opencast-admin-interface] Preselect all capture agent inputs when scheduling new event (PR #979)


Hi Greg,

Yes, we still use galicaster.

Thanks & Regards,
Nadia Malek.
From: Greg Logan ***@***.***>
Sent: 18 November 2024 23:35
To: opencast/opencast-admin-interface ***@***.***>
Cc: Nadia Malek ***@***.***>; Mention ***@***.***>
Subject: Re: [opencast/opencast-admin-interface] Preselect all capture agent inputs when scheduling new event (PR #979)


@JamesUoM [github.com]<https://urldefense.com/v3/__https:/github.com/JamesUoM__;!!PDiH4ENfjr2_Jw!AzI53ToNo0gESJRxIH9oIC99O4aSFwC0W_F-5-sJMrxvcefLcTqOI-0ksvoAeR8X1wwrZveF1spFqsmsQnUwHGuXSCBDbivd$> @NadiaUoM [github.com]<https://urldefense.com/v3/__https:/github.com/NadiaUoM__;!!PDiH4ENfjr2_Jw!AzI53ToNo0gESJRxIH9oIC99O4aSFwC0W_F-5-sJMrxvcefLcTqOI-0ksvoAeR8X1wwrZveF1spFqsmsQnUwHGuXSDswQyB0$> you guys still using Galicaster?

-
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.com/v3/__https:/github.com/opencast/opencast-admin-interface/pull/979*issuecomment-2484388788__;Iw!!PDiH4ENfjr2_Jw!AzI53ToNo0gESJRxIH9oIC99O4aSFwC0W_F-5-sJMrxvcefLcTqOI-0ksvoAeR8X1wwrZveF1spFqsmsQnUwHGuXSD4s4efZ$>, or unsubscribe [github.com]<https://urldefense.com/v3/__https:/github.com/notifications/unsubscribe-auth/BFC3B5YBSO7U6YHSBRJJBAD2BJ2TZAVCNFSM6AAAAABRTWWKAKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBUGM4DQNZYHA__;!!PDiH4ENfjr2_Jw!AzI53ToNo0gESJRxIH9oIC99O4aSFwC0W_F-5-sJMrxvcefLcTqOI-0ksvoAeR8X1wwrZveF1spFqsmsQnUwHGuXSKsgPZor$>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

—
Reply to this email directly, view it on GitHub [github.com]<https://urldefense.com/v3/__https://github.com/opencast/opencast-admin-interface/pull/979*issuecomment-2485322061__;Iw!!PDiH4ENfjr2_Jw!HXTOWJXAUFJpw0OYLvmzE3671_NUFzzN3mcPdPU4arwk_-3an4UEx5kfnOkfUR54E9DLwVYro-01s4a3e6x_idJOyD65dsGvCA$>, or unsubscribe [github.com]<https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AAWQ6E7FXOW2DNYO2KEBB432BMIIZAVCNFSM6AAAAABRTWWKAKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDIOBVGMZDEMBWGE__;!!PDiH4ENfjr2_Jw!HXTOWJXAUFJpw0OYLvmzE3671_NUFzzN3mcPdPU4arwk_-3an4UEx5kfnOkfUR54E9DLwVYro-01s4a3e6x_idJOyD5zk2-pyg$>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

gregorydlogan commented:
Can you guys take a quick poke and see if this change breaks things for you while you're still using GC?

JamesUoM commented:
> Can you guys take a quick poke and see if this change breaks things for you while you're still using GC?

Ok, though I doubt we'll still be using GC when we eventually deploy this code ;-)

JamesUoM commented:
> > Can you guys take a quick poke and see if this change breaks things for you while you're still using GC?
> 
> Ok, though I doubt we'll still be using GC when we eventually deploy this code ;-)

BTW I assume Zaragoza will be using GC

JamesUoM commented:
This will work fine with GC and matches what we have in the old Admin UI.

NOTE: GC can be configured to "hide" the inputs/tracks from OC and just returns `capture.device.names=defaults`  if a scheduler requests `capture.device.names=defaults` or `capture.device.names=` than all enabled tracks are recorded

gregorydlogan commented:
This doesn't seem to work for me.  I get the right UI elements, but nothing's selected?
[Screencast from 2025-04-02 16-32-53.webm](https://github.com/user-attachments/assets/4b09b034-7d20-4cdc-87c6-48495820add4)

Arnei commented:
Might be bugged due to a faulty merge. However, as it is unclear if we even want this feature at all, I'll abstain from working on the code for now.

snoesberger commented:
I'd still love to have this feature. Btw. the behavior was already like that in the old admin UI.

From an end user's point of view, this improves usability and saves clicks. For Extron devices, the default with no active input is not a valid config because at least one input must be selected to make a usable recording.
And I fear that there will be many support cases because users forget to activate an input and end up with an empty recording.

priya-tik committed:
make menuPlacement configurable via props

priya-tik committed:
removed unwanted white space

Arnei committed:
Fix modal data not being loaded

One of the recent eslint PRs removed this by accident,
causing modals to not load their data (i.e. the event details not
loading metadata).
This fixes that.

Arnei committed:
Activate eslint camelcase

Turn on the the eslint rule camelcase.

Arnei committed:
Activate some eslints again

These rules were deactivated due to some unfortunate merging.
This activates them again.

Arnei committed:
Activate  eslint spaced-comment

Turn on the eslint rule spaced-comment.

Arnei committed:
Activate eslint await-thenable

Turn on the eslint rule await-thenable.

Arnei committed:
Activate eslint no-unsafe-assignment

Enables the eslint no-unsafe-assignment.

Does not fix every problem this rule surfaces, i.e.
unchecked json parses, but it fixes some of them
and therefore at least constitutes some progress.

Arnei committed:
Activate eslint no-unsafe-call

Enables the eslint rule no-unsafe-call.

Arnei committed:
Activate eslint no-unsafe-argument

Enables the eslint rule no-unsafe-argument

Arnei committed:
Activate eslint no-unsafe-member-access

Enable the eslint rule no-unsafe-member-access.

Arnei committed:
Activate eslint no-unsafe-return

Enable the eslint rule no-unsafe-return

Arnei committed:
Remove IconButton component

The IconButton component can be replaced with the similar
ButtonLikeAnchor component. No need to have two very
similar components, so
this removes one of them.

Arnei committed:
Remove IconButton component

The IconButton component can be replaced with the similar
ButtonLikeAnchor component. No need to have two very
similar components, so
this removes one of them.

Arnei committed:
Introduce BaseButton component

Adds a new component "BaseButton". This can and
should be used as the default button component.

Arnei committed:
Replace fontAwesome icons with lucide icons

This brings us more in line with the other Opencast interfaces.

priya-tik committed:
Merge develop into event-details-comments-dropdown-fix and resolve conflicts

priya-tik committed:
Merge develop into event-details-comments-dropdown-fix and resolve conflicts

priya-tik committed:
resolve conflicts

Arnei committed:
Remove fontawesome

Replace more fontawesome icons with lucide icons
and remove all fontawesome css files.

Arnei committed:
Remove fontawesome

Replace more fontawesome icons with lucide icons
and remove all fontawesome css files.

Arnei committed:
Merge pull request #1310 from priya-tik/event-details-comments-dropdown-fix

Event details comments dropdown fix

dependabot[bot] committed:
Bump the minor-and-patch group with 10 updates

---
updated-dependencies:
- dependency-name: "@emotion/styled"
  dependency-version: 11.14.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: "@mui/material"
  dependency-version: 7.2.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: axios
  dependency-version: 1.10.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: i18next
  dependency-version: 25.3.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: react-i18next
  dependency-version: 15.5.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: react-router
  dependency-version: 7.6.3
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: "@types/lodash"
  dependency-version: 4.17.19
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: eslint
  dependency-version: 9.30.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: typescript-eslint
  dependency-version: 8.35.1
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: vitest
  dependency-version: 3.2.4
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Arnei committed:
Apply default actions to default roles for new events/series

Fixes #1354.

An admin can configure actions that are automatically
added to each role you add to an ACL in the ACL tab
for series and events. However, they were
not applied to default user role the ACL
is initialized with. This adds the actions to
default user role.

Arnei committed:
Sort workflows by newest submit date

Currently the list of workflows in the workflow is displayed
by some undefined sorting that results in the oldest workflows
being at the top of the list. This sorts the list so that the
newest workflows are at the top instead.

Arnei committed:
Merge pull request #1340 from Arnei/hotfix-modal-data-not-loading

Fix modal data not being loaded

Arnei committed:
Move errors to top in workflow tab

Moves the workflow warnings and errors to the
top of the workflow details page. Also directly display
the list of warnings/errors, since it is usually not that long.

Arnei committed:
Fix custom actions being lost on template switch for default user

When switching between ACL templates, we were keeping
the default user as intended, but the default user
lost their default actions (if configured). This fixes that.

Arnei committed:
Display latest wf operation in wf details tab

Instead of just presenting a link to the "All operations" subtab,
this adds a display for the latest running task.
The intention here is to give users a quick impression on the
state of the workflow without having to access the
wf operations subtab.

Arnei committed:
Make error table look nicer for no errors in wf tab

In the workflow tab in the event details, don't display
the table header for the errors and warnings table
in case there are no errors.

Arnei committed:
Open workflow details first

When clicking on the Workflow tab in the event details,
the user is presented with a list of workflows to select
from. Most of the time, the user will selected the latest
workflow. So this patch selects it for them automatically.
The list of workflows can still be accessed through the tab
hierarch navigation.

Arnei committed:
Add icons to wf operation status

This adds icons in front of the status text in the workflow
operation list. This is supposed to make the state of the
workflow more easily identifiable at a glance.

Arnei committed:
Fix wf tab loading display

Arnei committed:
Display operation on workflow error details page

With some additional information from the backend, we can display
the workflow operation during which an error occured
on the error details page for easy access to information
about the operation.

Requires backend PR https://github.com/opencast/opencast/pull/6888

Arnei committed:
Only display current wf operation if wf running

The workflow tab in the event details displays the current
workflow operation. This is now only displayed if the workflow
is still running, because for finished workflows this
is a waste of screen space.

Arnei committed:
Reduce code duplication in wf tab

Just some code cleanup and formatting

Arnei committed:
Fix  acl tab config settings not applying

The acl config defaults were not always fetched for the acl tab.
This should fix that.

Arnei committed:
Merge pull request #1358 from opencast/dependabot/npm_and_yarn/minor-and-patch-27f82d48c3

Bump the minor-and-patch group with 10 updates

dependabot[bot] committed:
Bump vite from 6.3.5 to 7.0.0

---
updated-dependencies:
- dependency-name: vite
  dependency-version: 7.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Arnei committed:
Merge pull request #1357 from opencast/dependabot/npm_and_yarn/vite-7.0.0

Bump vite from 6.3.5 to 7.0.0

Arnei committed:
Reduce duplicate code in filter cell

Some cells in a table set a filter when their content is clicked.
This adds a component for these types of cells.

Also removes an unused component.

priya-tik committed:
Remove READ column from ACL modal UI

Arnei committed:
Reduce duplicate code in main page

Adds a single main page component, so pages don't
have to import the different headers and footer again and again.
Also adds a table page component, since almost all of our
pages are just a big table.

gregorydlogan committed:
Merge branch 'aria-disabled' of Arnei/opencast-admin-interface into r/17.x

Pull request #1335
Fixes #1217
  Add aria-disabled to relevant buttons

gregorydlogan committed:
Merge branch 'unify-search-field' of Arnei/opencast-admin-interface into r/17.x

Pull request #1337
Fixes #1269
  Unify search container

gregorydlogan committed:
Merge branch 'group-await-calls' of Arnei/opencast-admin-interface into r/17.x

Pull request #1341

  Group async calls

gregorydlogan committed:
Merge branch 'dropdown-with-variablesizelist' of Arnei/opencast-admin-interface into r/17.x

Pull request #1343

  Make DropDown more performant for many options

gregorydlogan committed:
Merge branch 'i18n-disable-debug' of Arnei/opencast-admin-interface into r/17.x

Pull request #1351

  Disable i18n debug

gregorydlogan committed:
Merge branch 'indicate-main-menu-area' of klaraloreen/opencast-admin-interface into develop

Pull request #1318

  Added indicator for main menu

gregorydlogan committed:
Merge branch 'eslint-camelcase' of Arnei/opencast-admin-interface into develop

Pull request #1345

  Activate eslint camelcase

gregorydlogan committed:
Merge branch 'eslint-no-non-null-asserted-optional-chain' of Arnei/opencast-admin-interface into develop

Pull request #1350

  Activate eslint no-non-null-asserted-optional-chain

gregorydlogan committed:
Merge branch 'unbound-method' of Arnei/opencast-admin-interface into develop

Pull request #1353

  Activate eslint unbound-method

Arnei committed:
Merge remote-tracking branch 'upstream/r/17.x' into develop

Arnei committed:
Add back missing icon

Got lost in rebasing

Arnei committed:
Fix wrong parameter name

Arnei committed:
Wait for workflowId to be loaded

Arnei committed:
Fix ACL role dropdown width

The role dropdown in the ACL tabs could extend over its column width, due to its own fixed width.
This fixes that.

gregorydlogan committed:
Merge branch 'workflow-tab-rework' of Arnei/opencast-admin-interface into develop

Pull request #1362

  Mini workflow tab rework

gregorydlogan committed:
Merge branch 'eslint-activate-again' of Arnei/opencast-admin-interface into develop

Pull request #1347

  Activate some eslints again

gregorydlogan committed:
Fixing issues caused by merging something that wasn't ready

gregorydlogan committed:
Merge branch 't/fix-mismerge' of gregorydlogan/opencast-admin-interface into develop

Pull request #1371

  Fix merge conflicts

Commit:
Automatically update translation keys

Arnei committed:
Merge pull request #1352 from Arnei/eslint-no-unsafe-assignment

Activate eslint no-unsafe-anything

Arnei committed:
Merge pull request #1349 from Arnei/eslint-await-thenable

Activate eslint await-thenable

Arnei committed:
Remove non-Promise awaits

Arnei committed:
Merge pull request #1348 from Arnei/eslint-spaced-comment

Activate  eslint spaced-comment

Arnei committed:
Pacify typescript with generic

Arnei committed:
Merge branch 'develop' into capture-agent-inputs-preselect

Arnei committed:
Activate eslint require-await

Enable the eslint rule require-await.

Arnei committed:
Activate eslint no-unused-vars

Enable the eslint rule no-unused-vars

gregorydlogan committed:
Merge branch 'unify-buttons' of Arnei/opencast-admin-interface into r/17.x

Pull request #1355

  Unify buttons

priya-tik committed:
Restore READ column in ACL UI

Commit:
Automatically update translation keys

Arnei committed:
Make event metadata load faster by fetching series options asynchronously

If you have many series in your Opencast, any modal that displays
event metadata can take several seconds to load. This patch aims at
pushing the loading time back into millisecond terrritory no matter
how many series there are.

The problem is that when fetching event metadata from the backend,
the backend also fetch all series options, which can take too long.
So we change the backend to not do that, and change the frontend to
fetch the series options separately.

ideally we could also do this for other metadata fields with options
(notably contributors and presenters), but the endpoint lacks
the necessary filter capabilities.

EnsiyehE committed:
fix/1283-input-unselect

EnsiyehE committed:
fix/1283-input-unselect

Arnei committed:
Merge pull request #1370 from Arnei/acl-role-dropdown-width-17

Fix ACL role dropdown width

Commit:
Automatically update translation keys

Arnei committed:
Clicking on roles in user table to set filter

Allows users to click on a role in the user table to set
that role as filter.

EnsiyehE committed:
fix the tab issue inside add-event modal

Arnei committed:
Fix Metadata field getting cut off

Some metadata fields would not display their full content
even if there was still space available. This patch should help
with that.

Arnei committed:
Fix datepicker opening twice on start date

When creating an event, on the source tab, the datepicker
for the start date metadata field is opening again after clicking
on a date. The first click also does not change the value.
This fixes this somewhat crudely.

EnsiyehE committed:
second commit

EnsiyehE committed:
Final Fix for the confirmation modal and focus trap

Arnei committed:
Fix schedule datepicker not working

When scheduling an event, or editing a scheduled event, the datepicker
for the scheduled date would not allow users to select a date.
This patch fixes that.

Arnei committed:
Show unsaved changes warning on tab switch

When changing event/series metadata/acl in the event/series details
and not explicitly saving them, the changes are lost when switching to
another tab. This patch causes the browser to show a warning dialog
when a user tries to switch to a different tab while there are unsaved
changes.

Commit:
Automatically update translation keys

Commit:
Automatically update translation keys

Arnei committed:
More explanations for embedd modal

Fixes #1272.
This mostly adds additional screen reader text to the embedding code
modal, but also some tooltips and
an explanatory text on the modal itself.

Arnei committed:
Merge remote-tracking branch 'upstream/r/17.x' into r/18.x

Arnei committed:
Merge remote-tracking branch 'upstream/r/18.x' into develop

Arnei committed:
Merge branch 'develop' into reduce-duplicate-code-filtercells

Commit:
Automatically update translation keys

Arnei committed:
Fix crowdin upload branch name

Since we adapted the branching structure of the main opencast
repository we do not use main anymore. This patch adapts the
run command from the same workflow in the mian repo.

Arnei committed:
Merge pull request #1394 from Arnei/crowdin-upload-branch

Fix crowdin upload branch name

Arnei committed:
Merge remote-tracking branch 'upstream/r/17.x' into r/18.x

Arnei committed:
Merge remote-tracking branch 'upstream/r/18.x' into develop

EnsiyehE committed:
Resolve the keyboard-nav by pressing tab

EnsiyehE committed:
Merge branch 'develop' into fix/modal-keyboard-nav-950

dependabot[bot] committed:
Bump the minor-and-patch group across 1 directory with 16 updates

Bumps the minor-and-patch group with 15 updates in the / directory:

| Package | From | To |
| --- | --- | --- |
| [@mui/material](https://github.com/mui/material-ui/tree/HEAD/packages/mui-material) | `7.2.0` | `7.3.1` |
| [axios](https://github.com/axios/axios) | `1.10.0` | `1.11.0` |
| [i18next](https://github.com/i18next/i18next) | `25.3.0` | `25.3.2` |
| [react](https://github.com/facebook/react/tree/HEAD/packages/react) | `19.1.0` | `19.1.1` |
| [react-dom](https://github.com/facebook/react/tree/HEAD/packages/react-dom) | `19.1.0` | `19.1.1` |
| [@types/react-dom](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/react-dom) | `19.1.6` | `19.1.7` |
| [react-i18next](https://github.com/i18next/react-i18next) | `15.5.3` | `15.6.1` |
| [react-router](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router) | `7.6.3` | `7.7.1` |
| [react-select](https://github.com/JedWatson/react-select) | `5.10.1` | `5.10.2` |
| [yup](https://github.com/jquense/yup) | `1.6.1` | `1.7.0` |
| [@types/lodash](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/lodash) | `4.17.19` | `4.17.20` |
| [eslint](https://github.com/eslint/eslint) | `9.30.0` | `9.32.0` |
| [sass](https://github.com/sass/dart-sass) | `1.89.2` | `1.90.0` |
| [typescript](https://github.com/microsoft/TypeScript) | `5.8.3` | `5.9.2` |
| [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) | `7.0.3` | `7.1.0` |



Updates `@mui/material` from 7.2.0 to 7.3.1
- [Release notes](https://github.com/mui/material-ui/releases)
- [Changelog](https://github.com/mui/material-ui/blob/master/CHANGELOG.md)
- [Commits](https://github.com/mui/material-ui/commits/v7.3.1/packages/mui-material)

Updates `axios` from 1.10.0 to 1.11.0
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.10.0...v1.11.0)

Updates `i18next` from 25.3.0 to 25.3.2
- [Release notes](https://github.com/i18next/i18next/releases)
- [Changelog](https://github.com/i18next/i18next/blob/master/CHANGELOG.md)
- [Commits](https://github.com/i18next/i18next/compare/v25.3.0...v25.3.2)

Updates `react` from 19.1.0 to 19.1.1
- [Release notes](https://github.com/facebook/react/releases)
- [Changelog](https://github.com/facebook/react/blob/main/CHANGELOG.md)
- [Commits](https://github.com/facebook/react/commits/v19.1.1/packages/react)

Updates `react-dom` from 19.1.0 to 19.1.1
- [Release notes](https://github.com/facebook/react/releases)
- [Changelog](https://github.com/facebook/react/blob/main/CHANGELOG.md)
- [Commits](https://github.com/facebook/react/commits/v19.1.1/packages/react-dom)

Updates `@types/react-dom` from 19.1.6 to 19.1.7
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react-dom)

Updates `react-i18next` from 15.5.3 to 15.6.1
- [Changelog](https://github.com/i18next/react-i18next/blob/master/CHANGELOG.md)
- [Commits](https://github.com/i18next/react-i18next/compare/v15.5.3...v15.6.1)

Updates `react-router` from 7.6.3 to 7.7.1
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router@7.7.1/packages/react-router)

Updates `react-select` from 5.10.1 to 5.10.2
- [Release notes](https://github.com/JedWatson/react-select/releases)
- [Changelog](https://github.com/JedWatson/react-select/blob/master/docs/CHANGELOG.md)
- [Commits](https://github.com/JedWatson/react-select/compare/react-select@5.10.1...react-select@5.10.2)

Updates `yup` from 1.6.1 to 1.7.0
- [Release notes](https://github.com/jquense/yup/releases)
- [Changelog](https://github.com/jquense/yup/blob/master/CHANGELOG.md)
- [Commits](https://github.com/jquense/yup/compare/v1.6.1...v1.7.0)

Updates `@types/lodash` from 4.17.19 to 4.17.20
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/lodash)

Updates `@types/react-dom` from 19.1.6 to 19.1.7
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/react-dom)

Updates `eslint` from 9.30.0 to 9.32.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Changelog](https://github.com/eslint/eslint/blob/main/CHANGELOG.md)
- [Commits](https://github.com/eslint/eslint/compare/v9.30.0...v9.32.0)

Updates `sass` from 1.89.2 to 1.90.0
- [Release notes](https://github.com/sass/dart-sass/releases)
- [Changelog](https://github.com/sass/dart-sass/blob/main/CHANGELOG.md)
- [Commits](https://github.com/sass/dart-sass/compare/1.89.2...1.90.0)

Updates `typescript` from 5.8.3 to 5.9.2
- [Release notes](https://github.com/microsoft/TypeScript/releases)
- [Changelog](https://github.com/microsoft/TypeScript/blob/main/azure-pipelines.release-publish.yml)
- [Commits](https://github.com/microsoft/TypeScript/compare/v5.8.3...v5.9.2)

Updates `typescript-eslint` from 8.35.1 to 8.39.0
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.39.0/packages/typescript-eslint)

Updates `vite` from 7.0.3 to 7.1.0
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/create-vite@7.1.0/packages/vite)

---
updated-dependencies:
- dependency-name: "@mui/material"
  dependency-version: 7.3.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: axios
  dependency-version: 1.11.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: i18next
  dependency-version: 25.3.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: react
  dependency-version: 19.1.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: react-dom
  dependency-version: 19.1.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: "@types/react-dom"
  dependency-version: 19.1.7
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: react-i18next
  dependency-version: 15.6.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: react-router
  dependency-version: 7.7.1
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: react-select
  dependency-version: 5.10.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: yup
  dependency-version: 1.7.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: "@types/lodash"
  dependency-version: 4.17.20
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: "@types/react-dom"
  dependency-version: 19.1.7
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: minor-and-patch
- dependency-name: eslint
  dependency-version: 9.32.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: sass
  dependency-version: 1.90.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: typescript
  dependency-version: 5.9.2
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: typescript-eslint
  dependency-version: 8.39.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
- dependency-name: vite
  dependency-version: 7.1.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: minor-and-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Arnei committed:
Merge pull request #1373 from Arnei/eslint-no-unused-vars

Activate eslint no-unused-vars

Arnei committed:
Merge branch 'develop' into eslint-require-await

 Bitte geben Sie eine Commit-Beschreibung für Ihre Änderungen ein. Zeilen,

Arnei committed:
Merge pull request #1372 from Arnei/eslint-require-await

Activate eslint require-await

Arnei committed:
Replace icon.pngs with lucide icons

The logical follow-up to replacing fontawesome icons with
lucide icons.
We have couple of png files we use as img files. These are all be
replaced with lucide icons.
Obselete pngs and css are deleted.

Arnei committed:
Remove unused scss

Replacing fontawesome icons with lucide icons allowed us to get
rid of a ton of scss. So why not get rid of more scss? As far as I
could tell, all of the scss removed in this commit is currently
going unused is unlikely to ever see use again.

Arnei committed:
Fix series not loading

Apparently our textFilter is now automatically treated as wildcard
search and the "*" we used to trigger wildcard search are now breaking
it, so we can just remove them.

Arnei committed:
Fix wrong import

Arnei committed:
npx eslint --fix

Eslint autoformatting

Arnei committed:
Merge pull request #1397 from opencast/dependabot/npm_and_yarn/minor-and-patch-6bda306d83

Bump the minor-and-patch group across 1 directory with 16 updates

Arnei committed:
Remove "accordion-header"

We are not using the accordion capabiltiies anywhere.

Arnei committed:
Remove the "clearfix" mixin

Clearfix helps with certain "float" issues. In our code it was either
added unnecessarily, or the issue could be solved in another way easily
enough.

Arnei committed:
Fix sass warnings

Removes some css that does not go together with other css,
according to sass.

Arnei committed:
Remove unused css from main.scss

Removes lots of leftover corrections that we do not need anymore.

Arnei committed:
Unify table css

Arnei committed:
Shuffle css around

Move css to more appropriate places. Only movement, no changes.

Arnei committed:
Remove more stuff from main.scss

And either crete new files or put it in the appropriate places.

Arnei committed:
Condense modal css

integrate the css for modals from main to modal-base

Arnei committed:
Fix missing remove buttons

Forgot to add new icons for some of the remove buttons, should
be fixed now.

Arnei committed:
Replace hardcoded colors with color variables

EnsiyehE committed:
fix/multiselect fields by Tab navigation

Arnei committed:
Move color css from tsx to scss

We have a bunch of css concerning colors flying around in our
tsx files. To ensure our colors are consistent in the future, this patch
moves the css into our scss files and replaces hex codes with variables.

Arnei committed:
Move more style to scss files

While the use of the "style" attribute can be necessary in many cases,
we mostly seem to use it because we were too lazy to put the css in it's
proper place in the scss files.
Let's remedy that.

Arnei committed:
Move inline icons styles in header to scss

Arnei committed:
Center plus button

Arnei committed:
Align LuExternalLink

Arnei committed:
Better align "Details >" chevron

Arnei committed:
Adjust comment delete icon css

Arnei committed:
Make stop icon grey again

Arnei committed:
Use lucide hand instead of hero hand

Arnei committed:
Shrink down workflow tab icons

Arnei committed:
Center main menu icon

Arnei committed:
Center warning icon

Arnei committed:
Slightly adjust thank you message icon

Arnei committed:
Remove leftovers

Arnei committed:
Better center service icon

Arnei committed:
Fix leftover imports

Arnei committed:
Fix notification icon being misaligned

Arnei committed:
Remove dead css class names

These don't exist anymore, so we can remove them.

Arnei committed:
Merge pull request #1356 from Arnei/lucide-icons

Add Lucide icons

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into acl-actions-missing-for-default-role

Arnei committed:
Fix default role actions not showing up for default user

For the acls, default actions can be configured to be added
automatically to each new role. However,
they were not added to the default role that gets automatically added.
This should fix that by removing some unnecessary react state.

Arnei committed:
Fix certain dropdown opening in wrong places

EnsiyehE committed:
fix tabbing issue

EnsiyehE committed:
Merge remote-tracking branch 'origin/develop' into fix/modal-keyboard-nav-950

EnsiyehE committed:
test to fix the building error

EnsiyehE committed:
build error

snoesberger committed:
Use formik 'FastField' instead of 'Field' for performance reasons

On an Opencast system with a large number of series (>4,000), entering text in
the create or edit event dialogue box is slow. Using the formik 'FastField'
instead of 'Field' improves the performance.

Fixes https://github.com/opencast/opencast-admin-interface/issues/1406

gregorydlogan committed:
Merge branch 'datepicker-opening-twice' of Arnei/opencast-admin-interface into r/17.x

Pull request #1384
Fixes #1388
  Fix datepicker opening twice on start date

gregorydlogan committed:
Merge branch 'schedule-datepicker-not-working' of Arnei/opencast-admin-interface into r/17.x

Pull request #1389

  Fix schedule datepicker not working

gregorydlogan committed:
Merge branch 'unsaved-changes-warning-on-tab-switch' of Arnei/opencast-admin-interface into r/17.x

Pull request #1390
Fixes #1342
  Show unsaved changes warning on tab switch

gregorydlogan committed:
Merge branch 'reduce-duplicate-code-tablepage' of Arnei/opencast-admin-interface into r/18.x

Pull request #1369

  Reduce duplicate code in main page

gregorydlogan committed:
Merge branch 'users-role-filter-cells' of Arnei/opencast-admin-interface into r/18.x

Pull request #1382

  Clicking on roles in user table to set filter

ltekbas committed:
Added styling for Generated index.html

ltekbas committed:
Create index.css

KatrinIhler committed:
Fix appliance of series ACL to new event

Storage of common metadata for the frontend was changed, rename key
accordingly.

gregorydlogan committed:
Merge branch 'fix-series-rights-for-new-events' of KatrinIhler/opencast-admin-interface into r/17.x

Pull request #1416
Fixes #1410
  Fix appliance of series ACL to new event

gregorydlogan committed:
Revert "Fix appliance of series ACL to new event"

This reverts commit 60010c737e530f64f35821baf7f10c2f24fc9efd, which
breaks the UI when an event is added to a series

gregorydlogan committed:
Merge pull request #1417 from gregorydlogan/t/rollback-1416

Revert 1416

lkiesow committed:
Validate and reset corrupted table filter states

Investigating people complaining develop.opencast.org being broken, it
seems like there was a change to the filters which cannot handle the
previous data which might still be stored in browsers. Like here means,
it completely crashes the interface and users just see a blank page.

The error they get is:

```
Uncaught TypeError: t.find is not a function
    fk tableFilterSelectors.ts:23
    D reselect.mjs:647
    i reselect.mjs:584
    P reselect.mjs:659
    i reselect.mjs:584
    u TableFilters.tsx:55
    <anonymous> Redux
    L use-sync-external-store-with-selector.production.js:40
    h use-sync-external-store-with-selector.production.js:63
    React 2
    useSyncExternalStoreWithSelector use-sync-external-store-with-selector.production.js:74
    n Redux
    qc TableFilters.tsx:55
    React 14
    K scheduler.production.js:152
tableFilterSelectors.ts:23:32
```

This patch introduces a general validation for filters, resetting them
if they are corrupted in the sense that the data structure is not
correct.

This closes #1381
This closes #1395

gregorydlogan committed:
Merge branch 'validate-state' of lkiesow/opencast-admin-interface into r/18.x

Pull request #1418

  Validate and reset corrupted table filter states

KatrinIhler committed:
Fetch series ACL only when series changed

KatrinIhler committed:
Reapply "Fix appliance of series ACL to new event"

This reverts commit 5d4da4358ce5c5ebcdfa3d6b6de102600b8817e7.

KatrinIhler committed:
Merge pull request #1419 from KatrinIhler/fix-series-rights-for-new-events-again

Fix series rights for new events - properly this time

KatrinIhler committed:
Merge branch 'r/17.x' into r/18.x

KatrinIhler committed:
Fix merge error and checkstyle warnings

KatrinIhler committed:
Merge branch 'r/18.x' into develop

KatrinIhler committed:
Fix checkstyle error after merge

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into css-more-cleanup

Arnei committed:
Avoid table overflowing in modals

Some table fields in modals may overflow if they contain a
very long word, for example the metadata td's in the create event modal
on the summary page. This commit fixes that by specifying word break
bevahiour for td's.

Arnei committed:
Fix default actions not being applied on first load

Not my favourite way to solve things, but with our current setup, this
is the way to go.

Ensures we have fetched the acl defaults from the backend
before opening the create event/series modal, so that Formik can be
properly initialized with said defaults
without the need for enabling reinitialization.

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into acl-actions-missing-for-default-role

ltekbas committed:
added task to copy index.css to the web server

lkiesow committed:
Merge pull request #1408 from snoesberger/1406-event-textfield-optimization

Use formik 'FastField' instead of 'Field' for performance reasons

ltekbas committed:
moved css a folder down, edit href command

gregorydlogan committed:
Merge branch 'metadata-field-width' of Arnei/opencast-admin-interface into r/17.x

Pull request #1383

  Fix Metadata field getting cut off

gregorydlogan committed:
Merge branch 'more-text-for-embed-modal' of Arnei/opencast-admin-interface into r/17.x

Pull request #1391
Fixes #1272
  More explanations for embedd modal

gregorydlogan committed:
Merge branch 'async-select-for-collections' of Arnei/opencast-admin-interface into r/18.x

Pull request #1375

  Make event metadata load faster by fetching series options asynchronously

owi92 committed:
Ensure pasted values properly update path

This fixes an issue where pasting values instead
of typing them would lead to stale page data,
effectively ignoring the pasted value in path creation.
This is solved by explicitly adding the new segment to the
page object.

lkiesow committed:
Merge pull request #1376 from EnsiyehE/fix/1283-input-unselect

fix/1283-input-unselect

Arnei committed:
Default template for "Create Event/Series"

Includes #1360. #1360 should be reviewed and merged first.

This allows an admin to configure a default template for new
events/series. Upon opening the ACL tab of the respective create modal,
the template will already be selected and the roles respective to the
template will be set.

A default template will not overwrite the default user role of the
creator that is added automatically.

pselsa committed:
Merge remote-tracking branch 'origin/develop' into 296-terms-of-use

pselsa committed:
fix some type issues

Arnei committed:
Disable tooltips on tables due to performance issues

This commit disables tooltips for all table cells. For example,
when moving the mouse over the red X in the events table,
there will no longer be a tooltip showing. This is done to
imporve performance.

It appears that the mui tooltip component is not performant when
used in large quantities. According to this comment on github it is not
supposed to be:
https://github.com/mui/material-ui/issues/27057#issuecomment-880236030

I locally tested performance by profiling the rendering of the rows of the
series table. The table had 1000 entries which necessitated 4000 tooltip
components. Rendering the table with tooltips took about 3,7 seconds.
Rendering the table without tooltips took about 1,2 seconds. I think
this is way too much time for tooltips.

The goal of this commit is to be a quick fix for quick gains.
A more proper long term solution would let us keep the tooltips, by for
example working around the issue somehow or replacing mui tooltips with
a different library.

Arnei committed:
Don't render series modal in action cells

This commit removes the series details modal component from every
action cell in the table, and instead adds the component to the
respective central page file (i.e. Series.tsx). The modal can then
be opened through redux, similar to how it already works for events.

The goal of these changes is to speed up table rendering.
For a series table with 1000 rows, these changes sped up rendering by
about 150ms in my local testing.

This also fixes an issue where opening the events details modal would
cause the entire events table to rerender. Opening the events details
should be a bit smoother now.

Arnei committed:
Remove unnecessary html tags from table cells

Removes html tags that are not needed from table cell
components. This won't really make any sort of noticable dent
performance wise, but it still results in a cleaner dom, so why not.

Arnei committed:
Avoid unnecessary table rerenders

This patch prevents certain actions from causing all table rows
to rerender, if the actions did not actually effect the table rows.
It does this by refactoring the table component so that redux state
updates only trigger rerenders in the
components that actually use the update data.

Actions include:
- Selecting a single row.
- Opening the page size dropdown.
- Displaying a tooltip by hovering over a hoverable table element.

Arnei committed:
Use template name to fetch default template

Assumes that:
- Instead of id, "default_template" is now the templates name.
- There is a new endpoint `/admin-ng/acl/acl/${name}`

Arnei committed:
Don´t run corrupt state check on state change

The goal of useTableFilterStateValidation
is to fix state when switching between ui versions.
We do not need to fix state while the app is running,
Therefore I think it is reasonable to run it only once on load, in order
to reduce rerenders.

ltekbas committed:
Edit index.html command, added css command

gregorydlogan committed:
Merge branch '296-terms-of-use' of schuettloeffel-elsa/opencast-admin-interface into develop

Pull request #825

  Optional terms of use dialog for user

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into css-more-cleanup

KatrinIhler committed:
Fix series ACLs not being applied to new events

gregorydlogan committed:
Merge branch 'r/17.x' of manschwa/opencast-admin-interface into r/17.x

Pull request #1430

  Backport fix/1283-input-unselect  into branch r/17.x

gregorydlogan committed:
Merge branch 'remove-read-colum-fix' of priya-tik/opencast-admin-interface into develop

Pull request #1368

  Remove READ column from ACL modal UI

gregorydlogan committed:
Merge branch 'fix/modal-keyboard-nav-950' of EnsiyehE/opencast-admin-interface into develop

Pull request #1387

  Fix/modal keyboard nav 950

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into reduce-duplicate-code-filtercells

Arnei committed:
Remove unused imports

miesgre committed:
Improvement: Update the submitter display in the event details tab

ltekbas committed:
removed index changes from test file, edit assets command before checkout and edit genrate index html

ltekbas committed:
added missing $

owi92 committed:
Correct indentation of large block

KatrinIhler committed:
Merge branch 'r/18.x' into develop

miesgre committed:
Use a Tooltip to display the submitter name in the event details tab

JulianKniephoff committed:
Take advantage of all available space for dropdowns in metadata modals

JulianKniephoff committed:
Take advantage of all available space for dropdowns in metadata modals

JulianKniephoff committed:
Refactor dropdown components and fix width in metadata forms

This is a follow-up to #1444 where we fixed dropdown width for r/17.x.
Due to the refactoring in #1375, we need to apply the same fix for r/18.x.

gregorydlogan committed:
Merge branch 'disable-tooltips-on-tables' of Arnei/opencast-admin-interface into r/17.x

Pull request #1433

  Disable tooltips on tables due to performance issues

gregorydlogan committed:
Merge branch 'dont-render-series-modal-in-action-cells' of Arnei/opencast-admin-interface into r/17.x

Pull request #1435

  Don't render series modal in action cells

gregorydlogan committed:
Merge branch 'series-selection-width' of JulianKniephoff/opencast-admin-interface into r/17.x

Pull request #1444

  Take advantage of all available space for dropdowns in metadata modals

gregorydlogan committed:
Merge branch 'add-username-on-event-details-workflow-tab' of miesgre/opencast-admin-interface into r/17.x

Pull request #1443

  Improvement: Update the submitter display in the event details tab

gregorydlogan committed:
Merge branch 'tobira-paste-bug' of owi92/opencast-admin-interface into r/18.x

Pull request #1428

  Fix Tobira paste bug

gregorydlogan committed:
Merge branch 'dont-run-state-validation-on-every-filter-change' of Arnei/opencast-admin-interface into r/18.x

Pull request #1438
Fixes #1432
  Don´t run corrupt state check on state change

gregorydlogan committed:
Merge branch 'fix-series-acl' of KatrinIhler/opencast-admin-interface into r/18.x

Pull request #1440

  Fix series ACLs not being applied to new events

gregorydlogan committed:
Merge branch 'series-selection-width-2' of JulianKniephoff/opencast-admin-interface into r/18.x

Pull request #1445

  Refactor dropdown components and fix width in metadata modals

Arnei committed:
Merge remote-tracking branch 'upstream/r/17.x' into r/18.x

Arnei committed:
Merge remote-tracking branch 'upstream/r/18.x' into develop

Arnei committed:
Merge branch 'develop' into reduce-duplicate-code-filtercells

Arnei committed:
Merge branch 'develop' into css-more-cleanup

dependabot[bot] committed:
Bump uuid from 11.1.0 to 13.0.0

Bumps [uuid](https://github.com/uuidjs/uuid) from 11.1.0 to 13.0.0.
- [Release notes](https://github.com/uuidjs/uuid/releases)
- [Changelog](https://github.com/uuidjs/uuid/blob/main/CHANGELOG.md)
- [Commits](https://github.com/uuidjs/uuid/compare/v11.1.0...v13.0.0)

---
updated-dependencies:
- dependency-name: uuid
  dependency-version: 13.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump camelcase-keys from 9.1.3 to 10.0.0

Bumps [camelcase-keys](https://github.com/sindresorhus/camelcase-keys) from 9.1.3 to 10.0.0.
- [Release notes](https://github.com/sindresorhus/camelcase-keys/releases)
- [Commits](https://github.com/sindresorhus/camelcase-keys/compare/v9.1.3...v10.0.0)

---
updated-dependencies:
- dependency-name: camelcase-keys
  dependency-version: 10.0.0
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @types/node from 22.15.18 to 24.6.1

Bumps [@types/node](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/HEAD/types/node) from 22.15.18 to 24.6.1.
- [Release notes](https://github.com/DefinitelyTyped/DefinitelyTyped/releases)
- [Commits](https://github.com/DefinitelyTyped/DefinitelyTyped/commits/HEAD/types/node)

---
updated-dependencies:
- dependency-name: "@types/node"
  dependency-version: 24.6.1
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump @vitejs/plugin-react-swc from 3.10.2 to 4.1.0

Bumps [@vitejs/plugin-react-swc](https://github.com/vitejs/vite-plugin-react/tree/HEAD/packages/plugin-react-swc) from 3.10.2 to 4.1.0.
- [Release notes](https://github.com/vitejs/vite-plugin-react/releases)
- [Changelog](https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react-swc/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite-plugin-react/commits/v4.1.0/packages/plugin-react-swc)

---
updated-dependencies:
- dependency-name: "@vitejs/plugin-react-swc"
  dependency-version: 4.1.0
  dependency-type: direct:development
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/checkout from 4 to 5

Bumps [actions/checkout](https://github.com/actions/checkout) from 4 to 5.
- [Release notes](https://github.com/actions/checkout/releases)
- [Changelog](https://github.com/actions/checkout/blob/main/CHANGELOG.md)
- [Commits](https://github.com/actions/checkout/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/checkout
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump actions/setup-node from 4 to 5

Bumps [actions/setup-node](https://github.com/actions/setup-node) from 4 to 5.
- [Release notes](https://github.com/actions/setup-node/releases)
- [Commits](https://github.com/actions/setup-node/compare/v4...v5)

---
updated-dependencies:
- dependency-name: actions/setup-node
  dependency-version: '5'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
Bump softprops/turnstyle from 2 to 3

Bumps [softprops/turnstyle](https://github.com/softprops/turnstyle) from 2 to 3.
- [Release notes](https://github.com/softprops/turnstyle/releases)
- [Changelog](https://github.com/softprops/turnstyle/blob/master/CHANGELOG.md)
- [Commits](https://github.com/softprops/turnstyle/compare/v2...v3)

---
updated-dependencies:
- dependency-name: softprops/turnstyle
  dependency-version: '3'
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

TIKaushik committed:
Fix overlapping buttons in “Series details → Access policy” dialog #1426 (#1439)

* #1422 fixed the button alignment.

* opencast#1422 fixed the button alignment.

Arnei committed:
Fix workflow tab not showing for scheduled events

Fixes #1459.

The workflow tab in the event details of scheduled events was
not showing. It should now show again.

Arnei committed:
Merge pull request #1451 from opencast/dependabot/npm_and_yarn/uuid-13.0.0

Bump uuid from 11.1.0 to 13.0.0

Arnei committed:
Merge pull request #1452 from opencast/dependabot/npm_and_yarn/camelcase-keys-10.0.0

Bump camelcase-keys from 9.1.3 to 10.0.0

Arnei committed:
Merge pull request #1454 from opencast/dependabot/npm_and_yarn/vitejs/plugin-react-swc-4.1.0

Bump @vitejs/plugin-react-swc from 3.10.2 to 4.1.0

Arnei committed:
Merge pull request #1453 from opencast/dependabot/npm_and_yarn/types/node-24.6.1

Bump @types/node from 22.15.18 to 24.6.1

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into css-more-cleanup

Arnei committed:
Render date field in scheduling tab as input field

Fixes #1374.

The start date field in the scheduling tab of the event modals
is rendering like a metadata field. This makes it look like it cannot
be edited. This changes it so that the date field always looks like an
input field, which is the way it was before.

gregorydlogan committed:
Fixing issues with upstream PR:
 - Saving assets ahead of checkout of gh-pages
 - Updating assets after checkout, if appropriate (develop only)
 - Added missing logo

gregorydlogan committed:
Merge branch 'ltekbas-generateIndexHTML' of ltekbas/opencast-admin-interface into r/17.x

Pull request #1415

  Landing page suitable for non-technical users

gregorydlogan committed:
Merge branch 'slightly-cleaner-table-cells' of Arnei/opencast-admin-interface into r/17.x

Pull request #1436

  Remove unnecessary html tags from table cells

gregorydlogan committed:
Merge branch 'refactor-table' of Arnei/opencast-admin-interface into r/17.x

Pull request #1437

  Avoid unnecessary table rerenders

gregorydlogan committed:
Merge branch 'workflow-scheduling-tab-does-not-show-in-event-details' of Arnei/opencast-admin-interface into r/18.x

Pull request #1460
Fixes #1459
  Fix workflow tab not showing for scheduled events

gregorydlogan committed:
Merge branch 'scheduling-date-field-should-look-more-ready' of Arnei/opencast-admin-interface into r/17.x

Pull request #1463
Fixes #1374
  Render date field in scheduling tab as input field

gregorydlogan reviewed (changes requested):
Looks like the code underneath changed?

```
/home/greg/opencast/opencast-admin-interface/src/components/events/partials/ModalTabsAndPages/NewSourcePage.tsx
  627:15  error  'inputDevice' is never reassigned. Use 'const' instead  prefer-const
```

gregorydlogan committed:
Merge branch 'acl-actions-missing-for-default-role' of Arnei/opencast-admin-interface into develop

Pull request #1360
Fixes #1354
  Apply default actions to default roles for new events/series

gregorydlogan committed:
Merge branch 'dependabot/github_actions/actions/checkout-5' of opencast/opencast-admin-interface into develop

Pull request #1456

  Bump actions/checkout from 4 to 5

gregorydlogan committed:
Merge branch 'dependabot/github_actions/actions/setup-node-5' of opencast/opencast-admin-interface into develop

Pull request #1457

  Bump actions/setup-node from 4 to 5

gregorydlogan committed:
Merge branch 'dependabot/github_actions/softprops/turnstyle-3' of opencast/opencast-admin-interface into develop

Pull request #1458

  Bump softprops/turnstyle from 2 to 3

gregorydlogan committed:
Merge branch 'reduce-duplicate-code-filtercells' of Arnei/opencast-admin-interface into develop

Pull request #1367

  Reduce duplicate code in filter cell

gregorydlogan committed:
Merge branch 'css-more-cleanup' of Arnei/opencast-admin-interface into develop

Pull request #1400

  Css cleanup

gregorydlogan committed:
Merge branch 'default-create-template' of Arnei/opencast-admin-interface into develop

Pull request #1429

  Default ACL template for "Create Event/Series"

Arnei committed:
Merge remote-tracking branch 'upstream/r/17.x' into r/18.x

Arnei committed:
Eslint Missing trailing comma

Arnei committed:
Merge remote-tracking branch 'upstream/r/18.x' into develop

Arnei committed:
Merge remote-tracking branch 'upstream/develop' into capture-agent-inputs-preselect
